src/disc/stk/Albany_STKDiscretization.cpp

```diff
@@ -2024,12 +2029,71 @@ STKDiscretization::computeSideSets()
       lssList[ss_key].numSides       = globalSideSetViews[ss_key].max_sides;
       lssList[ss_key].numCellsOnSide = Kokkos::subview(globalSideSetViews[ss_key].numCellsOnSide, current_index, Kokkos::ALL() );
       lssList[ss_key].cellsOnSide    = Kokkos::subview(globalSideSetViews[ss_key].cellsOnSide,    current_index, Kokkos::ALL(), Kokkos::ALL() );
+      lssList[ss_key].sideSetIdxOnSide    = Kokkos::subview(globalSideSetViews[ss_key].sideSetIdxOnSide,    current_index, Kokkos::ALL(), Kokkos::ALL() );
 
       current_local_index[ss_key]++;
 
       ss_it++;
     }
   }
+
+  // 6) Populate localDOFViews for GatherVerticallyContractedSolution
+  const int maxSideNodes = 6;
+  const Albany::LayeredMeshNumbering<GO>& layeredMeshNumbering = *(stkMeshStruct->layered_mesh_numbering);
```

Comment:
I do actually check if layered_mesh_numbering exists but a few lines later for some reason. I'll move that up to avoid some unnecessary cycles and see if there is a way to check if it is needed by any evaluators.

Reply:
The latter might be a bit difficult. I don't see a clear parameter that would turn this on so it's probably okay to only construct this if the layered mesh numbering exists. It's probably not a lot of memory compared to the other memory issues we have.

Reply:
I don't think the disc has any way to know if the LMN is indeed needed.